.github/workflows/publish-fluentd-image.yaml

```diff
@@ -0,0 +1,153 @@
+            NEW_PATCH_VERSION=$((LATEST_PATCH_VERSION + 1))
+            IMAGE_BASE_TAG="${VERSION}-${NEW_PATCH_VERSION}"
+            echo "Building patch release ${IMAGE_BASE_TAG}!"
+          else
+            echo "${VERSION} tag does not exist, assuming we're building a new release!"
+            IMAGE_BASE_TAG="${VERSION}"
+          fi
+  
+          echo "IMAGE_BASE_TAG=$IMAGE_BASE_TAG" >> $GITHUB_OUTPUT
+          echo "IMAGE_MAJOR_MINOR=$MAJOR_MINOR" >> $GITHUB_OUTPUT
+
+  build:
+    name: Build/push image (${{ matrix.platform }})
+    runs-on: ${{ matrix.runs-on }}
+    needs: determine-tags
+    permissions:
+      actions: read
+      packages: write
+    outputs:
+      digest_amd64: ${{ steps.output-digests.outputs.amd64 }}
+      digest_arm64: ${{ steps.output-digests.outputs.arm64 }}
+      tags: ${{ steps.image-metadata.outputs.tags }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm64
+        include:
+          - runs-on: ubuntu-latest
+          - runs-on: ubuntu-24.04-arm # Builds arm64 on arm64 hosts 
+            platform: linux/arm64
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          platforms: linux/amd64,linux/arm64
+    
+      - name: Login to GHCR
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          registry: docker.io
+          username: ${{ secrets.REGISTRY_USER }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+
+      - name: Configure image tags 
+        id: image-metadata
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ env.GHCR_REPO }}
+            ${{ env.DOCKERHUB_REPO }}
+          flavor:
+            latest=false
+          tags: |
+            type=raw,value=latest
+            type=raw,value=${{ needs.determine-tags.outputs.IMAGE_BASE_TAG }}
+            type=raw,value=v${{ needs.determine-tags.outputs.IMAGE_BASE_TAG }}
+            type=raw,value=${{ needs.determine-tags.outputs.IMAGE_MAJOR_MINOR }}
+            type=raw,value=v${{ needs.determine-tags.outputs.IMAGE_MAJOR_MINOR }}
+
+      - name: Build and push image
+        id: build
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          file: cmd/fluent-watcher/fluentd/Dockerfile
+          platforms: ${{ matrix.platform }}
+          labels: ${{ steps.image-metadata.outputs.labels }}
+          provenance: false
+          outputs: type=image,"name=${{ env.GHCR_REPO }},${{ env.DOCKERHUB_REPO }}",push-by-digest=true,name-canonical=true,push=true
```

Comment:
This step produces arm64 and amd64 digests and pushes them to both GHCR and Docker Hub.